pkg/pagerduty/pagerduty_test.go

```diff
@@ -0,0 +1,151 @@
+			})
+		})
+
+		When("If sent input parameters are invalid", func() {
+			It("Should throw an error (400 badRequest)", func() {
+				//Arrange
+				mux.HandleFunc("/incidents", func(w http.ResponseWriter, r *http.Request) {
+					Expect(r.Method).To(Equal("PUT"))
+					w.Header().Set("Content-Type", "application/json")
+					w.WriteHeader(http.StatusBadRequest)
+					_, err := w.Write([]byte(fmt.Sprintf(`{"error":{"code":%d}}`, pagerduty.InvalidInputParamsErrorCode)))
+					Expect(err).NotTo(HaveOccurred())
+				})
+				// Act
+				err := p.MoveToEscalationPolicy(incident, escalationPolicy)
+				// Assert
+				Expect(err).Should(HaveOccurred())
+
+				Expect(err).Should(MatchError(pagerduty.InvalidInputParamsErr{}))
+
+			})
+		})
+
+		When("The Escalation policy has succesfully changed", func() {
+			It("Doesn't trigger an error", func() {
+				//Arrange
+				mux.HandleFunc("/incidents", func(w http.ResponseWriter, r *http.Request) {
+					Expect(r.Method).To(Equal("PUT"))
+					_, err := w.Write([]byte(`{"incidents": [{"title": "foo", "id": "1", "escalation_policy": {"id": "1234", "type": "escalation_policy_reference"}}]}`))
+					Expect(err).ShouldNot(HaveOccurred())
+				})
+				// Act
+				err := p.MoveToEscalationPolicy(incident, escalationPolicy)
+				// Assert
+				Expect(err).ShouldNot(HaveOccurred())
+				Expect(err).Should(BeNil())
+			})
+		})
+	})
+})
+
+/*
+ */
+
+func defaultTestClient(serverURL, authToken string) *sdk.Client {
+	c := sdk.NewClient(authToken,
+		sdk.WithAPIEndpoint(serverURL),
+		sdk.WithV2EventsAPIEndpoint(serverURL))
+	return c
+}
+
+// func newDefaultHTTPClient() *http.Client {
+// 	return &http.Client{
+// 		Transport: &http.Transport{
+// 			Proxy: http.ProxyFromEnvironment,
+// 			DialContext: (&net.Dialer{
+// 				Timeout:   30 * time.Second,
+// 				KeepAlive: 30 * time.Second,
+// 			}).DialContext,
+// 			MaxIdleConns:          10,
+// 			IdleConnTimeout:       60 * time.Second,
+// 			TLSHandshakeTimeout:   10 * time.Second,
+// 			ExpectContinueTimeout: 1 * time.Second,
+// 			MaxIdleConnsPerHost:   runtime.GOMAXPROCS(0) + 1,
+// 		},
+// 	}
+// }
+
+// HTTPClient is an interface which declares the functionality we need from an
+// HTTP client. This is to allow consumers to provide their own HTTP client as
+// needed, without restricting them to only using *http.Client.
+type HTTPClient interface {
+	Do(*http.Request) (*http.Response, error)
+}
+
+// defaultHTTPClient is our own default HTTP client. We use this, instead of
+// http.DefaultClient, to avoid other packages tweaks to http.DefaultClient
+// causing issues with our HTTP calls. This also allows us to tweak the
+// transport values to be more resilient without making changes to the
+// http.DefaultClient.
+//
+// Keep this unexported so consumers of the package can't make changes to it.
+//var defaultHTTPClient HTTPClient = newDefaultHTTPClient()
+
+//
+// func testMethod(t *testing.T, r *http.Request, want string) {
```

Comment:
Can we remove the unused function?

Reply:
Yes, I'll add a comment on where I pulled them from